### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by TheNeoGameFactory** [HaxeGodot/godot#1] [General]
Hi there, I heard a lot of haxe and how fast it is.

How it will perform against gdscript or c#?
More and better type programming is nice.

**Comment by ibilon** [comment]
While Haxe can target many languages, in order to be compatible with Godot this library only uses the C# target.
So the performances will be the same as using directly C#, and will compare the same to GDScript.

**Reply by TheNeoGameFactory** [reply]
Ah okay, thank you. ☺️
Are there some other benefits from using haxe? At the moment I only use gdscript but I know c# from unity.

Your examples with get_node typing, was this c# vs haxe?
If yes, then I like more the haxe syntax. So maybe I give it a try.

**Reply by ibilon** [reply]
`get_node` is available in GDScript, C# and Haxe.

What I've added is the typing of signals, where you use variables and function directly, instead of using strings.

Basically the main benefit of Haxe in this situation is the access to the Haxe ecosystem, in addition to the C# ecosystem.
So mainly a pro for people already doing Haxe.

But Haxe is a nice language with some powerful compile time features like abstracts and macros, among other things.
See this presentation by @nadako if you want to know more about Haxe https://slides.com/nadako/haxe/
